web-share/share-empty.https.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>WebShare Test: Share no known fields</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="resources/manual-helper.js"></script>
+  </head>
+  <body>
+    <script>
+        promise_test(t => {
+          return promise_rejects(t, new TypeError(), navigator.share());
+        }, 'share with no arguments (same as empty dictionary)');
+
+        promise_test(t => {
+          return promise_rejects(t, new TypeError(), navigator.share({}));
```

Comment:
@domenic, sorry to drag you in - but the above doesn't feel right to me (or, at least, I don't think this is how we've done extensibility in the platform before). Do you have an opinion?